src/log/CustomLog.cc

```diff
@@ -8,4 +8,9 @@
 
 #include "squid.h"
 #include "CustomLog.h"
+#include "log/File.h"
 
+bool CustomLog::usesDaemon() const
+{
+    return (logfile && logfile->flags.usesDaemon);
```

Comment:
AFAICT, `logfile` is set in accessLogInit() and icapLogOpen(). Both are called long after the new checks in configDoConfigure() are performed. Thus, the new checks will work incorrectly for CustomLog logs.

Reply:
Sigh. Good catch there, thank you. Redesigned to use just the config filename strings in CustomLog objects instead of the delayed File objects.